x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -176,7 +188,13 @@ func (k Keeper) CheckMisbehaviour(ctx sdk.Context, misbehaviour ibctmtypes.Misbe
 		return errorsmod.Wrapf(ibcclienttypes.ErrInvalidMisbehaviour, "invalid misbehaviour for client-id: %s", misbehaviour.ClientId)
 	}
 
-	// TODO check misb valset signatures here
+	// VerifyClientMessage calls verifyMisbehaviour which verifies that the headers in the misbehaviour
```

Comment:
In the previous comment where you have:
```

	// CheckMisbehaviour verifies that the headers have both he same block height and
	// different blockID hashes
```
you can remove the "headers have the same block height" because you do the check before as well. In this case we only call `CheckMisbehaviour` to verify that `blockID`s hashes are different.

Reply:
Updated in [fda0b0f8461](https://github.com/cosmos/interchain-security/pull/1401/commits/fda0b0f8461ff02cd05318061c39ea046c9cbee4).